Source/UnitEx.cpp

```diff
@@ -1129,14 +1138,19 @@ void UnitEx::SetAnimation(int animIdx, int animDelay, int animStartDelay, int bo
 	func(&(*unitArray)[unitID], 0, animIdx, animDelay, animStartDelay, boolInvisible, boolSkipDoDeath);
 }
 
+// HFL initialization must be checked before calling GetBeaconData
+BeaconData& GetBeaconData(int unitID)

```

Comment:
This function and GetLabData could be moved inside the class as private members and the unit ID pass removed. I think this makes sense, but took a break from programming before accomplishing.

Reply:
That's a good point. Sounds like a reasonable optional change.

I don't know that it needs to be made private.

Reply:
Since LabData is defined in UnitEx.cpp, I don't think I can make the function a member function without moving the definition from the .cpp file to the header. I'll just leave as is for now.